### DISCUSSION TITLE
Terraform integration without having to install podman itself

### DISCUSSION THREAD

**Discussion by SebTardif** [containers/podman#17832] [General]
### Feature request description

Docker has this -> https://registry.terraform.io/providers/kreuzwerker/docker/latest/docs/resources/container

Which allow to run Terraform step inside a docker container. But that requires to pre-install docker.

Since podman is in golang, and Terraform provider are also in golang. It could be super cool if we had also a provider for podman, and even more cool if doesn't require any podman install.

Why someone would want to use a container to execute something when in Terraform?

Well, beside about having for outcome a 'daemon'  / 'web site' as the main task. I also don't like to have to rely on some 'local' installation of tool. For example, if my Terraform script that has nothing to do with running a container need to execute some tool, linux package, I don't want to pre-install them in the system where Terraform execute. I would prefer I can rely 100% on container, so that there is no extra tool to install beside Terraform itself.

In other words, I may not have a provider to build a xyz. Building xyz need like 10 different tools. I would like one my Terraform step/resource to start a container doing this. Another case, is that Terraform function/expression is not able to generate a config file I need, maybe I can do it with Velocity or other template engine. Then Terraform could call a container with these tools, and the console output is the result I'm looking for.

### Suggest potential solution

Provide Terraform provider with podman 100% integrated, not requiring any pre-podman install.

### Have you considered any alternatives?

Using the provider for docker with a pre install of Docker is the current alternative.

**Comment by Luap99** [comment]
I am not sure what do you expect from us? We are not going to build third party integrations, we simply do not have any resources to do things like that.

Also statically including podman may be possible but this is very hard to get right and we do not provide a stable go API.

The easiest thing to do would be just point the integration at the podman socket, since we support the same docker API there it should just work. And if you find differences in the API you can report them as bugs here.

**Comment by rhatdan** [comment]
Moving this to a discussion.
